templates/opsmanager/conf-mms.properties.erb

```diff
@@ -0,0 +1,122 @@
+mongodb.ssl.PEMKeyFile=<%=@pem_key_file%>
+mongodb.ssl.PEMKeyFilePassword=<%@pem_key_password%>
+
+# #####################################
+# Kerberos Module (Optional)
+#
+# The following parameters are for configuring Ops Manager to use Kerberos to connection
+# to its MongoDB backing stores.
+#
+# jvm.java.security.krb5.conf: This should be the path to the Kerberos conf file. The value will be set to JVM's
+# java.security.krb5.conf.
+#
+# jvm.java.security.krb5.kdc: This should be the IP/FQDN of the KDC server. The value will be set to JVM's
+# java.security.krb5.kdc.
+#
+# jvm.java.security.krb5.realm: This is the default REALM for Kerberos. It is being used for JVM's
+# java.security.krb5.realm.
+#
+# mms.kerberos.principal: The principal we used to authenticate with MongoDB. This should be the exact same user
+# on the mongoUri above.
+#
+# mms.kerberos.keyTab: The absolute path to the keytab file for the principal.
+#
+# mms.kerberos.debug: The debug flag to output more information on Kerberos authentication process.
+#
+# Please note, all the parameters are required for Kerberos authentication, except jvm.java.security.krb5.conf and
+# mms.kerberos.debug. The mechanism will not be functioning if any of the setting value is missing.
+#
+# Assume your kdc server FQDN is kdc.example.com, your Kerberos default realm is: EXAMPLE.COM,
+# the host running Ops Manager is mmsweb.example.com, the Kerberos for Ops Manager is mms/mmsweb.example.com@EXAMPLE.com,
+# And you have a keytab file for mms/mmsweb.example.com@EXAMPLE.COM located at /path/to/mms.keytab, then the
+# configurations would be:
+#       jvm.java.security.krb5.kdc=kdc.example.com
+#       jvm.java.security.krb5.realm=EXAMPLE.COM
+#       mms.kerberos.principal=mms/mmsweb.example.com@EXAMPLE.COM
+#       mms.kerberos.keyTab=/path/to/mms.keytab
+#       mms.kerberos.debug=false
+#
+# ####################################
+jvm.java.security.krb5.conf=
+jvm.java.security.krb5.kdc=
+jvm.java.security.krb5.realm=
+mms.kerberos.principal=
+mms.kerberos.keyTab=
+mms.kerberos.debug=
+
+# #####################################
+# Instance Parameter Overrides
+#
+# In this section include any parameters to be used on this instance
+# of Ops Manager. These parameters will override any global configuration
+# stored in the Ops Manager database.
+#
+# See https://docs.opsmanager.mongodb.com/current/reference/configuration/
+# for additional information
+#
+# #####################################
+
+mms.ignoreInitialUiSetup=<%=@ignore_ui_setup%>
+mms.centralUrl=<%=@url%>:<%=$port%>
+mms.https.ClientCertificateMode=<%=@client_certificate_mode%>
+mms.fromEmailAddr=<%=@from_email_addr%>
+mms.replyToEmailAddr=<%=@reply_to_email_addr%>
+mms.adminEmailAddr=<%=@admin_email_addr%>
+mms.emailDaoClass=<%=@email_dao_class%>
+mms.mail.transport=<%=@mail_transport%>
+mms.mail.hostname=<%=@smtp_server_hostname%>
+mms.mail.port=<%=@smtp_server_port%>
+
+<%unless @user_svc_class.nil?%>
```

Comment:
or just `<%- if @user_svc_class -%>?`  The only things in puppet that are false, are `false` and `undef`, so `unless @foo.nil?` style is only needed for `Optional[Booleans]`s IIRC